pkg/apiclient/requester.go

```diff
@@ -0,0 +1,62 @@
+package apiclient
+
+import (
+	"fmt"
+	"github.com/OctopusDeploy/cli/pkg/constants"
+	"github.com/spf13/cobra"
+	"runtime/debug"
+	"strings"
+)
+
+type Requester interface {
+	GetRequester() string
+}
+
+type RequesterContext struct {
+	cmd *cobra.Command
+}
+
+type FakeRequesterContext struct {
+}
+
+func NewRequester(c *cobra.Command) *RequesterContext {
+	return &RequesterContext{
+		cmd: c,
+	}
+}
+
+func (r *FakeRequesterContext) GetRequester() string { return "octopus/0.0.0" }
+
+func (r *RequesterContext) GetRequester() string {
+	version := "0.0.0"
+	if info, ok := debug.ReadBuildInfo(); ok {
+		for _, dep := range info.Deps {
+			if dep.Path == "github.com/OctopusDeploy/cli" {
+				if dep.Version != "" {
+					version = dep.Version
+				}
+			}
+		}
+	}
+
+	if r.cmd == nil {
+		if version == "" {
+			return constants.ExecutableName
+		}
+		return fmt.Sprintf("%s/%s", constants.ExecutableName, version)
+	}
+
+	commands := []string{r.cmd.Name()}
+	var rootCmd string
+	parentCmd := r.cmd.Parent()
+	for parentCmd != nil {
+		name := parentCmd.Name()
+		if name == constants.ExecutableName && version != "" {
+			rootCmd = fmt.Sprintf("%s/%s", name, version)
+		} else {
+			commands = append([]string{name}, commands...)
+		}
+		parentCmd = parentCmd.Parent()
+	}
+	return fmt.Sprintf("%s (%s)", rootCmd, strings.Join(commands, ";"))
```

Comment:
Not sure if I'm interpreting this correctly, it looks like the user agent will be something like `octopus/1.0.0 (release;create)`

Is that right? If so, I'm just curious as to what the benefit is? It seems like it will make server-side parsing of the user agent a bunch harder

Reply:
this part is appended to the user agent generated by the go client, which you can see in the screenshots in the PR body

Reply:
We are restricted by RFC7230 in terms of the delimiters we use. In this case, we use parentheses (comments), which is allowed by the specification.